src/index.d.ts

```diff
@@ -141,4 +141,63 @@ declare module "react-native-popup-menu" {
     SlideInMenu: Function;
     Popover: Function;
   }>;
+
+  /**
+   * withMenuContext
+   * @see https://github.com/instea/react-native-popup-menu/blob/master/doc/api.md#menuprovider
+   */
+  interface TriggerLayoutType {
+    x?: number;
+    y?: number;
+    width?: number;
+    height?: number;
+  }
+
+  interface OptionsLayoutType {
+    width?: number;
+    height?: number;
+  }
+
+  interface MenuEntry {
+    name: string;
+    instance: Menu;
+    triggerLayout?: TriggerLayoutType;
+    optionsLayout?: OptionsLayoutType;
+    optionsCustomStyles?: MenuOptionsCustomStyle;
+  }
+
+  interface MenuRegistry {
+    subscribe: (instance: Menu) => void;
+    unsubscribe: (instance: Menu) => void;
+    updateLayoutInfo: (
+      name: string,
+      layouts?: {
+        triggerLayout?: TriggerLayoutType;
+        optionsLayout?: OptionsLayoutType;
+      }
+    ) => void;
+    setOptionsCustomStyles: (name: string, optionsCustomStyles: MenuOptionsCustomStyle) => void;
+    getMenu: (name: string) => MenuEntry;
+    getAll: () => MenuEntry[];
+  }
+
+  interface MenuActions {
+    openMenu: (name: string) => Promise<void>;
+    closeMenu: () => Promise<void>;
+    toggleMenu: (name: string) => Promise<void>;
+    isMenuOpen: () => boolean;
+  }
+
+  export interface MenuContext {
+    menuRegistry: MenuRegistry;
```

Comment:
hi, you are right, `menuRegistry` is not officially public. But maybe we can just comment it out here and keep for future if we see suitable to make it public

Reply:
Alright